lib/Data/Validate/Sanctions.pm

```diff
@@ -81,61 +81,64 @@ sub get_sanctioned_info { ## no critic (RequireArgUnpacking)
 
     my $data = $self->_load_data();
 
-    # prepare list of possible variants of names: LastnameFirstname and FirstnameLastname
-    my @full_name = ($first_name, $last_name || ());
-    
-    my @name_variants = map {
-        my $name = uc(join('.*', map { my $x = $_; $x =~ s/[[:^alpha:]]//g; $x } @$_));
-        $name
-    } ([@full_name], @full_name > 1 ? [reverse @full_name] : ());
-    
+    # Sub to remove non-alphabets from the name
+    my $clean_names = sub {
+
+        my ($full_name) = @_;
+
+        # Remove non-alphabets
+        my @cleaned_full_name = map { s/[^[:alpha:]\s]/ /gr } split ' ', $full_name;
+
+        # Remove trailing and leading whitespaces
+        @cleaned_full_name = map { s/^\s*(.*)\s*$/$1/g; split ' ', uc  } @cleaned_full_name;
```

Comment:
this does not work as expected. Sorry for missing this the last round.

```
# perl -le 'print ">" . (s/^\s*(.*)\s*$/$1/gr) . "<" for (@ARGV)' '   fasd     ' fads
>fasd     <
>fads<
```

As you can see, leading spaces are removed but trailing spaces are kept. The reason is the `*` in `.*` is greedy. It should work as expected if you change the regex to `^\s*(.*?)\s*$`. `*?` is the non-greedy version.

Reply:
Maybe you want to change this to

```
# perl -le 'print ">$_<" for (map { split q{ }, uc s/^\s*(.*?)\s*$/$1/gr } @ARGV)' '   fasd fa fad     ' fads
>FASD<
>FA<
>FAD<
>FADS<
```

This might even resolve the perlcritic complaint.

Reply:
Good catch. I'd suggest leaving out the regex entirely, since there isn't going to be any leading/trailing whitespace anyway:

```
perl -le 'print ">$_<" for (map { split " ", uc } @ARGV)' '   fasd fa fad
```

This would also be simple enough to combine with the previous step so that you're doing

```
my @cleaned_full_name = split " ", $full_name =~ s/[^[:alpha:]\s]/ /gr;
```

which should help avoid ending up with empty elements in the array, e.g. `Fréderico é Estrela` or `Alan Ó Néill`.

(search for `awk` in https://perldoc.pl/functions/split for details)

Reply:
@tfoertsch you didn't miss it, I changed it together with your other review, I tried to implement it with `gr` but it was giving me the message `Useless use of non-destructive substitution (s///r) in void context`, I didn't see the testing fails, I don't know why it passed on QA.

Thanks @tfoertsch @binarytom